core/packages/contracts/src/NativeTokens.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import "openzeppelin/access/Ownable.sol";
+import "openzeppelin/access/AccessControl.sol";
+import "openzeppelin/token/ERC20/extensions/IERC20Metadata.sol";
+
+import "./TokenVault.sol";
+import "./SubstrateTypes.sol";
+import "./NativeTokensTypes.sol";
+import "./OutboundChannel.sol";
+
+/// @title Native Tokens
+/// @dev Manages locking, unlocking ERC20 tokens in the vault. Initializes ethereum native
+/// tokens on the substrate side via create.
+contract NativeTokens is AccessControl {
+    /// @dev Describes the type of message.
+    enum Action {
+        Unlock
+    }
+
+    /// @dev Message format.
+    struct Message {
+        /// @dev The action type.
+        Action action;
+        /// @dev The message payload.
+        bytes payload;
+    }
```

Comment:
type could go in some kinda PeeredApp base class, as mentioned in https://github.com/Snowfork/snowbridge/pull/781/files#r1155020081 ?